### PR TITLE
Embed backend docker command

### PR DESCRIPTION
**Pull request by jt-dd**
Adding backend docker command to remove `kubehound.sh` bash script.

```
./bin/build/kubehound backend --help
Handle the kubehound stack - docker compose based stack for kubehound services (mongodb, graphdb and UI)

Usage:
  kubehound backend [command]

Available Commands:
  down        Tear down the kubehound stack
  reset       Restart the kubehound stack
  up          Spawn the kubehound stack
  wipe        Wipe the persisted backend data

Flags:
  -f, --file strings   Compose configuration files
  -h, --help           help for backend

Global Flags:
      --debug   Enable debug logs

Use "kubehound backend [command] --help" for more information about a command.
```

Also, the backend gets automatically spawned when using the basic command `./kubehound`.

Also, added an hidden command for all the spawning of the backend for the dev env:

```
./bin/build/kubehound dev --help
[devOnly] Spawn the kubehound dev stack for the system-tests (build from dockerfile)

Usage:
  kubehound dev [flags]
  kubehound dev [command]

Available Commands:
  system-tests [devOnly] Spawn the kubehound testing stack for the system-tests

Flags:
      --down   Tearing down the kubehound dev stack and deleting the data associated with it
  -h, --help   help for dev
      --ui     Tearing down the kubehound dev stack and deleting the data associated with it

Global Flags:
      --debug   Enable debug logs

Use "kubehound dev [command] --help" for more information about a command.
```

Added also a way to dump the local version : `kubehound version`

### CODE REVIEW

File: deployments/kubehound/embed.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import (
+	"embed"
+)
+
+const (
+	DefaultComposePath = "docker-compose.embed.yaml"
+)
+
+//go:embed docker-compose.embed.yaml
+var F embed.FS
```
Comment (edznux-dd) [inline deployments/kubehound/embed.go:12]: Just a note, as I'm not sure exactly how you want to handle this but:
you can use `var F string` directly, and avoid the need to readfile / do the error checks  / convert to byte slice/string.

This also removes the need for the `DefaultComposePath`.

(if you want to provide another file, dynamically, you'll also need to change the FS, but that can be done on the caller  or here, depends)

File: pkg/backend/containers.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		},
+		Start: api.StartOptions{
+			Wait:  true,
+			Watch: true,
+		},
+	})
+}
+
+func (b *Backend) Down(ctx context.Context) error {
+	log.I.Infof("Tearing down the kubehound stack")
```
Comment (edznux-dd) [inline pkg/backend/containers.go:92]:
```suggestion
	log.I.Info("Tearing down the kubehound stack")
```

**Review by edznux-dd — COMMENTED** [review]
Couple question / notes:
- You haven't deleted the kubehound-ingestor directory, is that intended?
- If an env is already running (or something is already listening on a port), it's spitting out the "help", which I don't think is helpful. IMO there should be a prompt asking to recreate the env. (We should probably ask before doing the action, in case there's some data that the user don't want to loose)
- Could you add a print of the context that kubehound without any argument will run in? With enviroment with multiple context, this will get confusing if we are not very explicit about the env that we are pulling the data from by default.


On my machine, the janusgraph is failling on:
```
 ✘ Container kubehound-release-kubegraph-1  Error                                                                                                                                                                                                       66.4s 
 ✔ Container kubehound-release-ui-1         Started                                                                                                                                                                                                      0.5s 
 ✔ Container kubehound-release-mongodb-1    Created                                                                                                                                                                                                      0.0s 
Error: dependency failed to start: container kubehound-release-kubegraph-1 is unhealthy

waiting for JanusGraph Server...
/etc/opt/janusgraph/janusgraph-server.yaml will be used to start JanusGraph Server in foreground
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/opt/janusgraph/lib/log4j-slf4j-impl-2.20.0.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/opt/janusgraph/lib/logback-classic-1.2.11.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
17:08:46 INFO  org.janusgraph.graphdb.server.JanusGraphServer.printHeader -                                                                       
   mmm                                mmm                       #     
     #   mmm   m mm   m   m   mmm   m"   "  m mm   mmm   mmmm   # mm  
     #  "   #  #"  #  #   #  #   "  #   mm  #"  " "   #  #" "#  #"  # 
     #  m"""#  #   #  #   #   """m  #    #  #     m"""#  #   #  #   # 
 "mmm"  "mm"#  #   #  "mm"#  "mmm"   "mmm"  #     "mm"#  ##m#"  #   # 
                                                         #            
                                                         "            

17:08:46 INFO  com.jcabi.log.Logger.infoForced - 0 attributes loaded from 330 stream(s) in 39ms, 104 saved, 5179 ignored: []
17:08:46 INFO  org.janusgraph.graphdb.server.JanusGraphServer.printHeader - JanusGraph Version: 1.0.0-rc2
17:08:46 INFO  org.janusgraph.graphdb.server.JanusGraphServer.printHeader - TinkerPop Version: 3.6.2
17:08:46 INFO  org.janusgraph.graphdb.server.JanusGraphServer.start - Configuring JanusGraph Server from /etc/opt/janusgraph/janusgraph-server.yaml
17:08:46 INFO  org.apache.tinkerpop.gremlin.server.util.MetricManager.addJmxReporter - Configured Metrics JmxReporter configured with domain= and agentId=
17:08:46 INFO  org.apache.commons.beanutils.FluentPropertyBeanIntrospector.introspect - Error when creating PropertyDescriptor for public final void org.apache.commons.configuration2.AbstractConfiguration.setProperty(java.lang.String,java.lang.Object)! Ignoring this property.
17:08:47 INFO  org.janusgraph.graphdb.idmanagement.UniqueInstanceIdRetriever.getOrGenerateUniqueInstanceId - Generated unique-instance-id=ac1113031-b9e1c66ac9961
17:08:47 INFO  org.janusgraph.diskstorage.Backend.getIndexes - Configuring index [search]
17:08:47 INFO  org.janusgraph.diskstorage.configuration.ExecutorServiceBuilder.buildFixedExecutorService - Initiated fixed thread pool of size 40
17:08:47 INFO  org.janusgraph.graphdb.database.StandardJanusGraph.<init> - Gremlin script evaluation is disabled
17:08:47 ERROR org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor.<init> - Could not invoke constructor on class org.janusgraph.graphdb.management.JanusGraphManager (defined by the 'graphManager' setting) with one argument of class Settings
17:08:47 ERROR org.janusgraph.graphdb.server.JanusGraphServer.lambda$main$0 - JanusGraph Server was unable to start and will now begin shutdown
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor.<init>(ServerGremlinExecutor.java:97) ~[gremlin-server-3.6.2.jar:3.6.2]
```

I have 5GB disk available, could do some cleanup but that seems large? 
If not, we should say so somewhere, 5GB in VM and stuff it's a bunch


Will do another pass tomorow but LGTM overall :)

File: cmd/kubehound/dev.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		return Backend.Down(ctx)
+	}
+
+	return Backend.BuildUp(ctx)
+}
+
+func init() {
+	envCmd.AddCommand(envTestingCmd)
+	envCmd.PersistentFlags().BoolVar(&downTesting, "down", false, "Tearing down the kubehound dev stack and deleting the data associated with it")
+	envCmd.Flags().BoolVar(&uiTesting, "ui", false, "Tearing down the kubehound dev stack and deleting the data associated with it")
```
[the remark this answers is not in the document]
Comment (jt-dd) [reply cmd/kubehound/dev.go]: done

File: pkg/backend/containers.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+				Pull:    true,
+			},
+			Services:      b.project.ServiceNames(),
+			Recreate:      api.RecreateForce,
+			RemoveOrphans: true,
+		},
+		Start: api.StartOptions{
+			Wait:  true,
+			Watch: true,
+		},
```
[the remark this answers is not in the document]
Comment (jt-dd) [reply pkg/backend/containers.go:87]: I think it is enough as we wait on Kubehound binary side anyway.

File: pkg/backend/project.go [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+)
+
+var (
+	DefaultReleaseComposePaths = []string{"docker-compose.yaml", "docker-compose.release.yaml"}
+	DefaultDatadogComposePath  = "docker-compose.datadog.yaml"
+)
+
+func loadProject(ctx context.Context, composeFilePaths []string) (*types.Project, error) {
+	var project *types.Project
+	var err error
```
[the remark this answers is not in the document]
Comment (jt-dd) [reply pkg/backend/project.go:25]: I dont like this type of annotation (not used anywhere else in KubeHound)

File: pkg/backend/containers.go (hunk 2 above)
Comment (jt-dd) [reply pkg/backend/containers.go]: done

File: cmd/kubehound/dev.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+	DefaultComposeTestingPath = []string{"./deployments/kubehound/docker-compose.yaml", "./deployments/kubehound/docker-compose.testing.yaml"}
+	DefaultComposeDevPath     = []string{"./deployments/kubehound/docker-compose.yaml", "./deployments/kubehound/docker-compose.dev.yaml"}
+	DefaultComposeDevPathUI   = "./deployments/kubehound/docker-compose.ui.yaml"
+	DefaultDatadogComposePath = "./deployments/kubehound/docker-compose.datadog.yaml"
+)
+
+var (
+	envCmd = &cobra.Command{
+		Use:    "dev",
+		Hidden: true,
```
[the remark this answers is not in the document]
Comment (jt-dd) [reply cmd/kubehound/dev.go:21]: adding in the Jira card to track it for later (when new release is official).

File: cmd/kubehound/version.go [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+var (
+	versionCmd = &cobra.Command{
+		Use:   "version",
+		Short: "print Kubehound version",
+		Long:  `print the current version of Kubehound`,
+		Run: func(cobraCmd *cobra.Command, args []string) {
+			log.I.Infof("kubehound version: %s (%s/%s)", config.BuildVersion, config.BuildArch, config.BuildOs)
+		},
```
[the remark this answers is not in the document]
Comment (jt-dd) [reply cmd/kubehound/version.go:16]: done

File: cmd/kubehound/dev.go [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+		Short:  "[devOnly] Spawn the kubehound testing stack",
+		Long:   `[devOnly] Spawn the kubehound dev stack for the system-tests (build from dockerfile)`,
+		PersistentPreRunE: func(cobraCmd *cobra.Command, args []string) error {
+			var err error
+			Backend, err = docker.NewBackend(cobraCmd.Context(), composePath)
+
+			return err
```
[the remark this answers is not in the document]
Comment (jt-dd) [reply cmd/kubehound/dev.go]: added the singleton as we discussed

File: pkg/backend/containers.go [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+	"github.com/docker/docker/api/types/container"
+)
+
+type Backend struct {
+	project        *types.Project
+	composeService api.Service
+	dockerCli      *command.DockerCli
+}
+
+func NewBackend(ctx context.Context, composeFilePaths []string) (*Backend, error) {
```
[the remark this answers is not in the document]
Comment (jt-dd) [reply pkg/backend/containers.go]: done

**Review by edznux-dd — APPROVED** [review]
LFG! :heavy_check_mark:

**Pull request merged**